src/test/regress/expected/view_propagation.out

```diff
@@ -0,0 +1,383 @@
+ obj_identifier
+---------------------------------------------------------------------
+(0 rows)
+
+SELECT * FROM (SELECT pg_identify_object_as_address(classid, objid, objsubid) as obj_identifier from pg_catalog.pg_dist_object) as obj_identifiers where obj_identifier::text like '%opt_prop_view%';
+ obj_identifier
+---------------------------------------------------------------------
+(0 rows)
+
+-- Drop a column that view depends on
+ALTER TABLE view_table_1 DROP COLUMN val_1 CASCADE;
+NOTICE:  drop cascades to 3 other objects
+DETAIL:  drop cascades to view prop_view_1
+drop cascades to view prop_view_3
+drop cascades to view prop_view_8
+-- Since prop_view_3 depends on the view_table_1's val_1 column, it should be dropped
+SELECT * FROM (SELECT pg_identify_object_as_address(classid, objid, objsubid) as obj_identifier from pg_catalog.pg_dist_object) as obj_identifiers where obj_identifier::text like '%prop_view_3%';
+ obj_identifier
+---------------------------------------------------------------------
+(0 rows)
+
+-- Drop a table that view depends on
+DROP TABLE view_table_2 CASCADE;
+NOTICE:  drop cascades to 2 other objects
+DETAIL:  drop cascades to view prop_view_2
+drop cascades to constraint f_key_for_local_table on table view_table_3
+NOTICE:  drop cascades to constraint f_key_for_local_table_1410200 on table view_prop_schema.view_table_3_1410200
+CONTEXT:  SQL statement "SELECT citus_drop_all_shards(v_obj.objid, v_obj.schema_name, v_obj.object_name, drop_shards_metadata_only := false)"
+PL/pgSQL function citus_drop_trigger() line XX at PERFORM
+NOTICE:  removing table view_prop_schema.view_table_3 from metadata as it is not connected to any reference tables via foreign keys
+-- Since prop_view_2 depends on the view_table_2, it should be dropped
+SELECT * FROM (SELECT pg_identify_object_as_address(classid, objid, objsubid) as obj_identifier from pg_catalog.pg_dist_object) as obj_identifiers where obj_identifier::text like '%prop_view_2%';
+ obj_identifier
+---------------------------------------------------------------------
+(0 rows)
+
+-- Show that unsupported CREATE OR REPLACE VIEW commands are catched by PG on the coordinator
+CREATE TABLE table_to_test_unsup_view(id int, val1 text);
+SELECT create_distributed_table('table_to_test_unsup_view', 'id');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE VIEW view_for_unsup_commands AS SELECT * FROM table_to_test_unsup_view;
+CREATE OR REPLACE VIEW view_for_unsup_commands(a,b) AS SELECT * FROM table_to_test_unsup_view;
+ERROR:  cannot change name of view column "id" to "a"
+HINT:  Use ALTER VIEW ... RENAME COLUMN ... to change name of view column instead.
+CREATE OR REPLACE VIEW view_for_unsup_commands AS SELECT id FROM table_to_test_unsup_view;
+ERROR:  cannot drop columns from view
+DROP SCHEMA view_prop_schema_inner CASCADE;
+NOTICE:  drop cascades to 3 other objects
```

Comment:
maybe increase the min messages to prevent future flakyness.

Should we also re-sync metadata to one of the worker nodes just for completeness before dropping? That'd be imitating add node

Reply:
I guess this is the only remaining comment. We should at least consider doing the first one.